models/Document/PageSnippet.php

```diff
@@ -597,7 +597,8 @@ public function getUrl($hostname = null, $scheme = null)
             }
         }
 
-        $url = $scheme . $hostname . $this->getFullPath();
+        $url = $scheme . $hostname;
+        $url .= $this->getPrettyUrl() ?: $this->getFullPath();
```

Comment:
```suggestion
        if($this instanceof Page && $this->getPrettyUrl()) {
            $url .= $this->getPrettyUrl();
        } else {
            $url .= $this->getFullPath();
        }
```

Reply:
We need to check if `$this` is a `Page` because `Snippet` doesn't have the method `getPrettyUrl()`.